src/components/Table.tsx

```diff
@@ -0,0 +1,302 @@
+import {
+  Button,
+  Link,
+  Input,
+  Table as BaseTable,
+  PaginationProps,
+} from "@arco-design/web-react";
+import { ComponentImpl, implementRuntimeComponent } from "@sunmao-ui/runtime";
+import { css, cx } from "@emotion/css";
+import { Type, Static } from "@sinclair/typebox";
+import { FALLBACK_METADATA, getComponentProps } from "../sunmao-helper";
+import { TablePropsSchema } from "../generated/types/Table";
+import { useMemo, useRef, useState } from "react";
+import { sortBy } from "lodash-es";
+import {
+  LIST_ITEM_EXP,
+  LIST_ITEM_INDEX_EXP,
+  ModuleRenderer,
+} from "@sunmao-ui/runtime";
+
+const TableStateSchema = Type.Object({
+  selectedRows: Type.Optional(Type.Array(Type.Any())),
+  selectedItem: Type.Optional(Type.Any()),
+});
+
+type SortRule = {
+  field: string;
+  direction?: "ascend" | "descend";
+};
+
+type filterDropdownParam = {
+  filterKeys: string[];
+  setFilterKeys: (filterKeys: string[], callback?: () => void) => void;
+  confirm: () => void;
+};
+const TableImpl: ComponentImpl<Static<typeof TablePropsSchema>> = (props) => {
+  const { app, mergeState, customStyle, services, data } = props;
+
+  const { className, columns, pagination, ...cProps } =
+    getComponentProps(props);
+
+  const rowSelectionType: "checkbox" | "radio" | undefined =
+    cProps.rowSelectionType === "default" ? undefined : cProps.rowSelectionType;
+
+  const [selectedRowKeys, setSelectedRowKeys] = useState<number[]>([]);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [sortRule, setSortRule] = useState<SortRule>({
+    field: "name",
+    direction: undefined,
+  });
+  const [filterRule, setFilterRule] = useState();
+
+  const filteredData = useMemo(() => {
+    let filteredData = data || [];
+    if (filterRule) {
+      Object.keys(filterRule).forEach((colIdx) => {
+        const value = filterRule[colIdx][0];
+        filteredData = filteredData?.filter((row) =>
+          value ? row[colIdx].indexOf(value) !== -1 : true
+        );
+      });
+    }
+    return filteredData;
+  }, [data, filterRule]);
+
+  const sortedData = useMemo(() => {
+    if (!sortRule || !sortRule.direction) {
+      return filteredData;
+    }
+
+    const sorted = sortBy(filteredData, sortRule.field);
+    return sortRule.direction === "ascend" ? sorted : sorted.reverse();
+  }, [filteredData, sortRule]);
+
+  const { pageSize } = pagination;
+  const currentPageData = sortedData?.slice(
+    (currentPage - 1) * pageSize,
+    currentPage * pageSize
+  );
+
+  const clearState = () => {
+    mergeState({ selectedRows: [] });
+    mergeState({ selectedItem: undefined });
+  };
+  const inputRef = useRef(null);
+
+  for (const column of columns!) {
```

Comment:
Change React props directly is bad. May use `columns.map` to generate a new array to do that is better.